runtime-sdk/src/dispatcher.rs

```diff
@@ -365,13 +392,45 @@ impl<R: Runtime> transaction::dispatcher::Dispatcher for Dispatcher<R> {
                     mkvs,
                     &self.host_info,
                 );
+
             // Perform state migrations if required.
             R::migrate(&mut ctx);
 
+            // Prefetch.
+            let mut txs: Vec<Result<Transaction, RuntimeError>> = Vec::with_capacity(batch.len());
+            let mut prefixes: BTreeSet<Prefix> = BTreeSet::new();
+            for tx in batch.iter() {
+                let res = match Self::decode_tx(&mut ctx, tx) {
+                    Ok(tx) => {
+                        if prefetch_enabled {
+                            Self::prefetch_tx(&mut prefixes, tx.clone()).map(|_| tx)
+                        } else {
+                            Ok(tx)
+                        }
+                    }
+                    Err(err) => Err(RuntimeError {
```

Comment:
The error derive macro now provides a conversion to `RuntimeError` so you should be able to just do `err.into()` here.

Reply:
fixed